### PR TITLE
20/add setting configs

### PR DESCRIPTION
**Pull request by azizmashkour**
Add redux and config app setting: theming and language

### CODE REVIEW

File: src/actions/types.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const SET_CONFIG = 'SET_CONFIG';
+export const SET_LANGUAGE = 'SET_LANGUAGE';
+export const THEME_COLOR = 'THEME_COLOR';
```
Comment (juancarlosfarah) [inline src/actions/types.js]: If the other ones have `SET_` as a prefix, probably `THEME_COLOR` should be `SET_THEME_COLOR`, no?

File: src/components/Description.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  handleTitle,
+  showTitle,
+}) => (
+  <div className="Description-container">
+    <h2>{t('Description')}</h2>
+    <p>
+      {t('LabDescription')}
+    </p>
+    <p>
+      {t('SelectOrgane')}
```
Comment (juancarlosfarah) [inline src/components/Description.js]: Typo: `Organ`

File: src/components/Description.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import {
+  Button,
+  Col,
+  Row,
+} from 'reactstrap';
+
+const Description = ({
+  t,
+  themeColor,
+  handleTitle,
```
Comment (juancarlosfarah) [inline src/components/Description.js]: `handleTitle` should be `toggleTitle`, right? Because it is bound to an `onChange` handler.

File: src/components/LangBox.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withNamespaces } from 'react-i18next';
+import { Col, Row } from 'reactstrap';
+import Select from 'react-select';
+import { options } from '../config/options';
+
+const LangBox = ({
+  handleLang,
```
Comment (juancarlosfarah) [inline src/components/LangBox.js]: Again, `handleLang` seems to me like it should be `handleChangeLang`, as you put it against an `onChange` handler.

File: src/components/MainView.js [hunk 5]
```diff
@@ -1,33 +1,105 @@
...
-      t={t}
-      heartBeat={heartBeat}
-      pressure={pressure}
-      handleSection={handleSection}
-    />
-  </Row>
+  <div className="Main-container">
+    { showTitle ? (
+      <Row>
+        <Col md={12}>
```
Comment (juancarlosfarah) [inline src/components/MainView.js]: Here you use a Number for `md`, but below you use string. Be consistent.

File: src/components/MainView.js [hunk 6]
```diff
@@ -1,33 +1,105 @@
...
+          <h1 className="lab-title" style={{ backgroundColor: defaultColor }}>{t('Synchronization of the ovarian cycle')}</h1>
+        </Col>
+      </Row>
+    ) : ''
+    }
+    <Row className="Main">
+      <Col sm="8">
```
Comment (juancarlosfarah) [inline src/components/MainView.js]: Here you use a String for `sm`, but above you use a Number. Be consistent.

File: src/components/MainView.js [hunk 7]
```diff
@@ -1,33 +1,105 @@
...
+          showCarotidNerve={showCarotidNerve}
+          showHeringNerve={showHeringNerve}
+          showBulbNerve={showBulbNerve}
+          showParaNerve={showParaNerve}
+          showCordNerve={showCordNerve}
+          showSympaNerve={showSympaNerve}
+          showHeartNerve={showHeartNerve}
+        />
+      </Col>
+      <Col sm="4">
```
Comment (juancarlosfarah) [inline src/components/MainView.js]: Here you use a String for `sm`, but above you use a Number. Be consistent.

File: src/index.js [hunk 8]
```diff
@@ -1,16 +1,13 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { I18nextProvider } from 'react-i18next';
-import i18n from './i18n';
 import 'bootstrap/dist/css/bootstrap.css';
 import './index.css';
-import App from './App';
-import registerServiceWorker from './registerServiceWorker';
+import Root from './Root';
+import * as registerServiceWorker from './registerServiceWorker';
```
Comment (juancarlosfarah) [inline src/index.js]: Why did you change the import statement for `registerServiceWorker()`?

File: src/student/StudentView.js [hunk 9]
```diff
@@ -30,14 +25,67 @@ class StudentView extends Component {
...
+    const { showSympaNerve } = this.state;
+    this.setState({ showSympaNerve: !showSympaNerve });
+  }
+
+  handleHeartHover = () => {
+    const { showHeartNerve } = this.state;
+    this.setState({ showHeartNerve: !showHeartNerve });
+  }
+
+  handleTitle = () => {
```
Comment (juancarlosfarah) [inline src/student/StudentView.js]: I would actually change this to `toggleTitle`.

File: src/components/Description.js (hunk 3 above)
Comment (juancarlosfarah) [reply src/components/Description.js]: I just saw below that it indeed just toggles (show/hide) the title, so it should probably be called that.

**Comment by azizmashkour** [comment]
Thank you @juancarlosfarah
